npk/src/manifest.rs

```diff
@@ -458,17 +563,17 @@ seccomp:
         let manifest = Manifest::from_str(&manifest)?;
 
         assert_eq!(manifest.init, Some(PathBuf::from("/binary")));
-        assert_eq!(manifest.name, "hello");
+        assert_eq!(manifest.name.to_string(), "hello");
         let args = manifest.args.ok_or_else(|| anyhow!("Missing args"))?;
         assert_eq!(args.len(), 2);
-        assert_eq!(args[0], "one");
-        assert_eq!(args[1], "two");
+        assert_eq!(args[0].to_string(), "one");
+        assert_eq!(args[1].to_string(), "two");
 
         assert_eq!(manifest.autostart, Some(Autostart::Critical));
         let env = manifest.env.ok_or_else(|| anyhow!("Missing env"))?;
         assert_eq!(
-            env.get("LD_LIBRARY_PATH"),
-            Some("/lib".to_string()).as_ref()
+            env.get(&"LD_LIBRARY_PATH".to_string().try_into()?),
+            Some("/lib".to_string().try_into()?).as_ref()
```

Comment:
There are a couple of place where a 

`TryFrom<&str> for NotNullString` would be handy.

or even better a 

`impl<T> TryFrom<T> for NotNullString where T: ToString`

Reply:
The second version conflicted with a definition from core. I implemented the first variant.

Reply:
Ah. Yes both traits are upstream.